File: learnAPI/4a.py
import requests
import pandas as pd

data_kota = pd.DataFrame({
    'Kota' : ['Indonesia', 'Japan', 'China', 'Singapore', 'Malaysia']
})

for negara in data_kota['Kota']:
    url = "http://universities.hipolabs.com/search?country=" + str(negara)
    # url = f"http://universities.hipolabs.com/search?country={negara}"

    response = requests.get(url)
    
    if response.status_code == 200:
        data = response.json()
        
        df = pd.DataFrame(data)
        df['domains'] = df['domains'].apply(lambda x: ''.join(x))
        df['web_pages'] = df['web_pages'].apply(lambda x: ''.join(x))
        print(df)
        print()

File: learnAPI/4a.r
library(jsonlite)
library(dplyr)
library(httr)
library(tibble)
library(tidyr)

data_kota <- data.frame(
    Kota = c('Indonesia', 'Japan', 'China', 'Singapore', 'Malaysia')
)

for (negara in data_kota$Kota){
    url <- sprintf("http://universities.hipolabs.com/search?country=%s", negara)
    # url <- paste("http://universities.hipolabs.com/search?country=", negara, sep = '')
    # url <- "http://universities.hipolabs.com/search?name=binus&country=Indonesia"

    response <- GET(url)

    if (http_status(response)$category == "Success") {
        json_content <- content(response, "text")
        data <- fromJSON(json_content)

        df <- as.data.frame(data)
        print(head(df))
    }
}